File: fe/lib/models/category/category_model.dart
class CategoryModel {
  final String id;
  final String categoryCode;
  final String categoryName;
  final String? imageCategory;

  CategoryModel({
    required this.id,
    required this.categoryCode,
    required this.categoryName,
    this.imageCategory,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) {
    return CategoryModel(
      id: json['_id'] ?? '',
      categoryCode: json['categoryCode'] ?? '',
      categoryName: json['categoryName'] ?? '',
      imageCategory: json['imageCategory'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'categoryCode': categoryCode,
      'categoryName': categoryName,
      'imageCategory': imageCategory,
    };
  }
}

File: be/models/category.model.js
const db = require("../configs/db");

const categorySchema = new db.mongoose.Schema(
  {
    categoryCode: { type: String, required: true, unique: true },
    categoryName: { type: String, required: true },
    imageCategory: { type: String },
    deleteAt: { type: Date, default: null },
  },
  { collection: "categories", timestamps: true },
);

const categoryModel = db.mongoose.model("categoryModel", categorySchema);
module.exports = { categoryModel };
